src/Action/RefundUnitsAction.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sylius\RefundPlugin\Action;
+
+use Prooph\ServiceBus\CommandBus;
+use Prooph\ServiceBus\Exception\CommandDispatchException;
+use Sylius\RefundPlugin\Request\RefundUnitsRequest;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\Session\Session;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+
+final class RefundUnitsAction
+{
+    /** @var CommandBus */
+    private $commandBus;
+
+    /** @var Session */
+    private $session;
+
+    /** @var UrlGeneratorInterface */
+    private $router;
+
+    public function __construct(CommandBus $commandBus, Session $session, UrlGeneratorInterface $router)
+    {
+        $this->commandBus = $commandBus;
+        $this->session = $session;
+        $this->router = $router;
+    }
+
+    public function __invoke(Request $request): Response
+    {
+        try {
+            $this->commandBus->dispatch(RefundUnitsRequest::getCommand($request));
+
+            $this->session->getFlashBag()->add('success', 'sylius_refund.units_successfully_refunded');
```

Comment:
I'm not sure if that is a right place to add notification. Maybe we should create a notification projector that will be listening to the `UnitsRefunded` event?